backend/src/minitwit-api.js

```diff
@@ -17,168 +17,46 @@ function updateLatest(req) {
-  }
-});
 
-app.post('/msgs/:username', async (req, res) => {
-  try {
-    updateLatest(req);
-    notReqFromSimulator(req, res);
-
-    const user = await db.user.findOne({ where: { username: req.params.username } });
-    if (!user) return res.status(404).send({ error: 'User not found' });
-
-    await user.createMessage({ text: req.body.content });
-
-    return res.status(204).send();
-  } catch (err) {
-    return handleError(err, res);
-  }
-});
-
-app.get('/fllws/:username', async (req, res) => {
-  try {
-    updateLatest(req);
-    notReqFromSimulator(req);
-
-    const { no: noFollowers = 100 } = req.body;
-
-    const user = await db.user.findOne({ where: { username: req.params.username } });
-    if (!user) return res.status(404).send({ error: 'User not found' });
-
-    const followRes = await user.getFollow({ limit: noFollowers });
-    const follows = followRes.map((flw) => flw.username);
-
-    return res.send({ follows });
-  } catch (err) {
-    return handleError(err, res);
-  }
-});
-
-app.post('/fllws/:username', async (req, res) => {
-  try {
-    updateLatest(req);
-    notReqFromSimulator(req);
-
-    const user = await db.user.findOne({ where: { username: req.params.username } });
-    if (!user) return res.status(404).send({ error: 'User not found' });
-
-    const keys = Object.keys(req.body);
-
-    if (keys.includes('follow')) {
-      const followed = await db.user.findOne({ where: { username: req.body.follow } });
-
-      if (followed === null) throw new Error('User not found');
-
-      await user.addFollow(followed);
+  console.log(err);
+  console.log('--> sent 500');
+  return res.status(500).send({ error_msg: err.message });
+}
 
-      return res.status(204).send();
-    }
+function notReqFromSimulator(req) {
+  const token = req.header('Authorization');
+  if (token !== 'Basic c2ltdWxhdG9yOnN1cGVyX3NhZmUh') throw new Error('Unauthorized');
+}
 
-    if (keys.includes('unfollow')) {
-      const unfollowed = await db.user.findOne({ where: { username: req.body.unfollow } });
-      if (!unfollowed) return res.status(404).send({ error: 'User not found' });
+const authenticate = require('./routes/authenticate')(updateLatest, notReqFromSimulator, handleError);
+const messages = require('./routes/messages')(updateLatest, notReqFromSimulator, handleError);
+const follows = require('./routes/follows')(updateLatest, notReqFromSimulator, handleError);
 
-      await user.removeFollow(unfollowed);
+app.use('', authenticate);
+app.use('/msgs', messages);
+app.use('/fllws', follows);
 
-      return res.status(204).send();
-    }
+app.get('/latest', (req, res) => res.send({ latest }));
 
-    return res.status(400).send({ error: 'Neither the follow or the unfollow key was given in request' });
-  } catch (err) {
-    return handleError(err, res);
-  }
+const server = app.listen(port, () => {
+  console.log(`Server started on port: ${port}`);
 });
 
-db.sequelize.sync().then(async () => {
```

Comment:
What is the motivation for removing this synchronize call?
Actually, I am a bit confused what it does... The docs says that it creates the schemas, but with an empty database, the first call to `create` or `update` seems to create the schemas as well (just like in Entity Framework). 
Originally, I thought that this DB call would ensure that the DB was running and ready for read/write operations before serving any HTTP requests. So I am a bit nervous for removing this call when it is being run with the production database. The DB has over 120K entries in it right now, so it will take some time to initialize, after a deployment, and being able to handle read/writes.
I have tested it locally with existing data (2K entries), and it does not seem to have an effect, so maybe it is just my own fear 🙃

I do not require any changes, and I think that we should go with it, but we should be aware of this if we have to make changes to the schemas, or the first request to the simulator takes a long time.